src/main/java/com/mraof/minestuck/item/foods/SurpriseEmbryoItem.java

```diff
@@ -18,20 +19,21 @@ public SurpriseEmbryoItem(Properties properties)
 	{
 		super(properties);
 	}
-
-	protected void onItemUseFinish(ItemStack stack, World world, PlayerEntity player)
+	
+	@Override
+	public ItemStack onItemUseFinish(ItemStack stack, World world, LivingEntity player)
 	{
-		super.onItemUseFinish(stack, world, player);
-		if(!player.world.isRemote)
+		if(player instanceof PlayerEntity && !player.world.isRemote)
 		{
 			Random ran = new Random();
 			ItemStack[] items = new ItemStack[]{new ItemStack(Items.MELON_SLICE), new ItemStack(Items.STICK), new ItemStack(Items.EGG),
 					new ItemStack(Blocks.DIRT), new ItemStack(Blocks.PUMPKIN), new ItemStack(Blocks.COBBLESTONE)};
 			int num = ran.nextInt(items.length);
-			player.inventory.addItemStackToInventory(items[num].copy());
-			ITextComponent message = new TranslationTextComponent("item.surpriseEmbryo.message", items[num].getDisplayName());
+			((PlayerEntity) player).inventory.addItemStackToInventory(items[num].copy());
```

Comment:
This function returns false if the players inventory is full. You should probably do something related to that. (See the obsidian bucket for an example)